docs/devguide/roadmap.rst

```diff
@@ -1,135 +1,50 @@
 
-.. image:: ../images/swift-e-timeline_trimmed.png
-
-
-Sufficient capabilities to use Parsl in many common situations already exist.  This document indicates where Parsl is going;
-it contains a list of features that Parsl has or will have.  Features that exist today are marked in bold, with the release
-in which they were added marked for releases since 0.3.0. Help in providing any of the yet-to-be-developed capabilities is welcome.
-
-Features in preparation are documented via Github
+Features and improvements are documented via GitHub
 `issues <https://github.com/Parsl/parsl/issues>`_ and `pull requests <https://github.com/Parsl/parsl/pulls>`_.
 
 
-Core Functionality
+Code Maintenance
 ---------------------
 
-* **Parsl has the ability to execute standard python code and to asynchronously execute tasks, called Apps.**
-    * **Any Python function annotated with "@App" is an App.**
-    * **Apps can be Python functions or bash scripts that wrap external applications.**
-* **Asynchronous tasks return futures, which other tasks can use as inputs.**
-    * **This builds an implicit data flow graph.**
-* **Asynchronous tasks can execute locally on threads or as separate processes.**
-* **Asynchronous tasks can execute on a remote resource.**
-    * **libsubmit (to be renamed) provides this functionality.**
-    * **A shared filesystem is assumed; data staging (of files) is not yet supported.**
-* **The Data Flow Kernel (DFK) schedules Parsl task execution (based on dataflow).**
-* **Class-based config definition (v0.6.0)**
-* **Singleton config, and separate DFK from app definitions (v0.6.0)**
-* Class-based app definition
+* **Type Annotations and Static Type Checking**: Add static type annotations throughout the codebase and add typeguard checks.
+* **Release Process**: `Improve the overall release process <https://github.com/Parsl/parsl/issues?q=is%3Aopen+is%3Aissue+label%3Arelease_process>`_ to synchronize docs and code releases, automatically produce changelog documentation.
+* **Components Maturity Model**: Defines the `component maturity model <https://github.com/Parsl/parsl/issues/2554>`_ and tags components with their appropriate maturity level.
+* **Define and Document Interfaces**: Identify and document interfaces via which `external components <https://parsl.readthedocs.io/en/stable/userguide/plugins.html>`_ can augment the Parsl ecosystem.
+* **Distributed Testing Process**: All tests should be run against all possible schedulers, using different executors, on a variety of remote systems. Explore the use of containerized schedulers and remote testing on real systems.
 
-Data management
+New Features and Integrations
 ---------------
```

Comment:
> /home/runner/work/parsl/parsl/docs/devguide/roadmap.rst:22:Title underline too short.

says the test system.

Reply:
I updated the underline lengths. I'll check the tests to make sure we are good to go.